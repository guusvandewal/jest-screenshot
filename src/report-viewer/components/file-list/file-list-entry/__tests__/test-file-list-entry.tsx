import * as React from "react";
import { shallow } from "enzyme";
import { FileListEntry } from "..";

describe("FileListEntry", () => {
    it("looks as expected", () => {
        expect(shallow(
            <FileListEntry
                file={testResults.files[0]}
            />,
        )).toMatchSnapshot();
    });

    it("changes the selected file", () => {
        const element = shallow(<FileListEntry file={testResults.files[1]} />);
        expect(element.find("a").hasClass("is-active")).toBe(false);
        element.find("a").simulate("click");
        expect(element.find("a").hasClass("is-active")).toBe(true);
    });
});
