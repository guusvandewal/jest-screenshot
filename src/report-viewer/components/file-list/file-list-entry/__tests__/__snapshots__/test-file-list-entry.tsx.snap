// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileListEntry looks as expected 1`] = `
<li>
  <a
    className="is-active"
    onClick={[Function]}
  >
    src/__tests__/test-first.ts
  </a>
</li>
`;
